src/components/InteractionLayer/index.js

```diff
@@ -374,8 +377,14 @@ class InteractionLayer extends React.Component {
   };
 
   syncZoomingState = () => {
-    if (this.props.zoomable && !this.props.onAreaDefined) {
+    const { onAreaDefined, onDoubleClick, zoomable } = this.props;
+    if (zoomable && !onAreaDefined) {
       this.rectSelection.call(this.zoom.on('zoom', this.zoomed));
+      if (onDoubleClick) {
+        this.rectSelection.on('dblclick.zoom', () => {
+          onDoubleClick(d3.event);
```

Comment:
I really wanted to make this like other events where if this doesn't return `false`, then the zoom would happen. I couldn't figure that one out, though.

Any help on this would be appreciated!

Reply:
Am I missing something, why not to to just pass null to `dblclick.zoom` callback if `onDoubleClick` is set and just pass `onDoubleClick` method into Interaction layer rect React SyntheticEvent `onDoubleClick`?

Reply:
Done